Dependencies/UrlLib/Include/UrlLib/UrlLib.h

```diff
@@ -53,6 +53,10 @@ namespace UrlLib
 
         std::string_view ResponseString() const;
 
+        std::string ResponseHeader(const std::string& headerName) const;
```

Comment:
This contract is a bit off.

```suggestion
        std::optional<std::string> GetResponseHeader(const std::string& headerName) const;
```

If the response headers are not available yet or if the `headerName` doesn't exist, we need to be able to distinguish from a header that returned an empty string.

This is a function call and not a property access function, so it should have a `Get` prefix.

Also, in the `Open` method, I used `std::string` as input which is inconsistent with this. I'm always annoyed that C++ doesn't have a standard way of passing in strings. I keep flipflopping between what is best here. It seems like `const std::string&` reduces copying, so let's stick with this. Can you update `Open` to use `const std::string&`?

Reply:
Fixed